.gitignore

```diff
@@ -0,0 +1,81 @@
+######### cpp related #############
+
+# Prerequisites
+*.d
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+*.dll
+
+# Fortran module files
+*.mod
+*.smod
+
+# Compiled Static libraries
+*.lai
+*.la
+*.a
+*.lib
+
+# Executables
+*.exe
+*.out
+*.app
+
+###################################
+
+######### cmake related ###########
+CMakeLists.txt.user
+CMakeCache.txt
+CMakeFiles
+CMakeScripts
+Testing
+Makefile
+cmake_install.cmake
+install_manifest.txt
+compile_commands.json
+CTestTestfile.cmake
+_deps
+build
+
+###################################
+
+
+######### vscode related ##########
+
+.vscode/
```

Comment:
Why do we need the vscode thingy? Is it okay if someone does not use vscode?

Reply:
My vscode creates that temporary file, so I suppress it in `.gitignore` because I don't want to be tracked by git.